webapi/src/main/scala/org/knora/webapi/responders/v1/PermissionsResponderV1.scala

```diff
@@ -0,0 +1,853 @@
+            //_ = log.debug(s"getDefaultObjectAccessPermissionV1 - result: ${MessageUtil.toSource(permissionQueryResponse)}")
+
+            permissionQueryResponseRows: Seq[VariableResultsRow] = permissionQueryResponse.results.bindings
+
+            _ = if (permissionQueryResponseRows.isEmpty) {
+                throw NotFoundException(s"Permission '$permissionIri' not found")
+            }
+
+            groupedPermissionsQueryResponse: Map[String, Seq[String]] = permissionQueryResponseRows.groupBy(_.rowMap("p")).map {
+                case (predicate, rows) => predicate -> rows.map(_.rowMap("o"))
+            }
+            //_ = log.debug(s"getDefaultObjectAccessPermissionV1 - groupedResult: ${MessageUtil.toSource(groupedPermissionsQueryResponse)}")
+
+            hasPermissions = PermissionUtilV1.parsePermissions(groupedPermissionsQueryResponse.get(OntologyConstants.KnoraBase.HasPermissions).map(_.head), PermissionType.OAP)
+
+            defaultObjectAccessPermission = DefaultObjectAccessPermissionV1(iri = permissionIri, forProject = groupedPermissionsQueryResponse.getOrElse(OntologyConstants.KnoraBase.ForProject, throw InconsistentTriplestoreDataException(s"Permission $permissionIri has no project.")).head, forGroup = groupedPermissionsQueryResponse.get(OntologyConstants.KnoraBase.ForGroup).map(_.head), forResourceClass = groupedPermissionsQueryResponse.get(OntologyConstants.KnoraBase.ForResourceClass).map(_.head), forProperty = groupedPermissionsQueryResponse.get(OntologyConstants.KnoraBase.ForProperty).map(_.head), hasPermissions = hasPermissions)
+
+            result = DefaultObjectAccessPermissionForIriGetResponseV1(defaultObjectAccessPermission)
+
+        } yield result
+    }
+
+    /**
+      * Gets a single default object access permission identified by project and either group / resource class / property.
+      *
+      * @param projectIRI the project's IRI.
+      * @param groupIRI the group's IRI.
+      * @param resourceClassIRI the resource's class IRI
+      * @param propertyIRI the property's IRI.
+      * @return an optional [[DefaultObjectAccessPermissionV1]]
+      */
+    def defaultObjectAccessPermissionGetV1(projectIRI: IRI, groupIRI: Option[IRI], resourceClassIRI: Option[IRI], propertyIRI: Option[IRI]): Future[Option[DefaultObjectAccessPermissionV1]] = {
+        for {
+            a <- Future("")
+
+            // check if necessary field are not empty.
+            _ = if (projectIRI.isEmpty) throw BadRequestException("Project cannot be empty")
+
+            // check supplied parameters.
+            parametersSupplied = List(groupIRI, resourceClassIRI, propertyIRI).flatten.size
+            _ = if (parametersSupplied != 1) throw BadRequestException("Either groupIri or resourceClassIri or propertyTypeIri can be supplied")
+
+            sparqlQueryString <- Future(queries.sparql.v1.txt.getDefaultObjectAccessPermission(
```

Comment:
Since this is not the first line of the `for` comprehension, there is no need to construct a `Future` here.